addons/stock/stock.py

```diff
@@ -2488,7 +2489,21 @@ def action_done(self, cr, uid, ids, context=None):
                             self.action_done(cr, uid, [move.move_dest_id.id], context=context)
 
             self._update_average_price(cr, uid, move, context=context)
-            self._create_product_valuation_moves(cr, uid, move, context=context)
+            # Use SUPERUSER_ID, otherwise warehouse users would need to
+            # be given access to Finance objects.
+            # We need to force company_id to company_id of present user,
+            # because SUPERUSER_ID might have a different company.
+            local_context = context.copy()
+            if ((not uid == SUPERUSER_ID)
+            and ('force_company' not in local_context)):
+                users_model = self.pool.get('res.users')
+                user_record = users_model.read(
+                    cr, uid, uid, ['company_id'], context=context)
+                assert user_record, _('User %d does not exist') % uid
```

Comment:
@StefanRijnhart No. Just a case of defensive (or ueberdefensive?) programming. If you do think it is better to leave it out, I can leave it out.